vmdb/db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogbundleinitialization.rb

```diff
@@ -16,42 +16,70 @@
 #
 # Inputs: $evm.root['service_template_provision_task'].dialog_options
 #
+# array_entry example - Classification::1
+def vmdb_object_from_array_entry(entry)
+  model, id = entry.split("::")
+  $evm.vmdb(model, id.to_i) if model && id
+end
+
+def array_value(hash, hash_key, value)
+  if hash.key?(hash_key)
+    values = hash[hash_key]
+    values = "#{values},#{value}"
+    values[0] = "" if values[0] == ","
+  else
+    value
+  end
+end
 
 # Description: Look for service dialog variables in the root object that start with "dialog_option_[0-9]",
 def get_options_hash(dialog_options)
   # Setup regular expression for service dialog tags
-  options_regex = /^(dialog_option|dialog_tag)_(\d*)_(.*)/i
-  options_hash = {}
+  options_regex       = /^(dialog_option|dialog_tag)_(\d*)_(.*)/i
+  options_array_regex = /^(Array::dialog_option|Array::dialog_tag)_(\d*)_(.*)/i
+  options_hash        = {}
 
   # Loop through all of the options and build an options_hash from them
-  dialog_options.each do |k, v|
+  dialog_options.each do |key, value|
 
-    option_key = k.downcase.to_sym
-    if options_regex =~ k
+    option_key = key.downcase.to_sym
+    if options_regex =~ key
       sequence_id = Regexp.last_match[2].to_i
+      next if value.blank?
 
-      unless v.blank?
-        $evm.log("info", "Adding sequence_id:<#{sequence_id}> option_key:<#{option_key.inspect}> v:<#{v.inspect}> to options_hash")
-        if options_hash.key?(sequence_id)
-          options_hash[sequence_id][option_key] = v
-        else
-          options_hash[sequence_id] = {option_key => v}
+      $evm.log("info", "Adding seq_id:<#{sequence_id}> key:<#{option_key}> value:<#{value}>")
+      if options_hash.key?(sequence_id)
+        options_hash[sequence_id][option_key] = value
+      else
+        options_hash[sequence_id] = {option_key => value}
+      end
+    elsif options_array_regex =~ key
+      sequence_id = Regexp.last_match[2].to_i
+      option_key = $3
+      value.split(",").each do |entry|
+        vmdb_obj = vmdb_object_from_array_entry(entry.to_s)
+        next if vmdb_obj.blank?
+
+        options_hash[sequence_id] = {} unless options_hash.key?(sequence_id)
```

Comment:
This 'options_hash[sequence_id] = {} unless options_hash.key?(sequence_id)` line could be removed.